### PR TITLE
internal/charmstore: enhance get upload info parts.

### PR DESCRIPTION
**Pull request by fabricematrat**

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/charmstore/2090/

**Review by jrwren — APPROVED** [review]

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/charmstore/2091/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/charmstore/2092/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/charmstore/2093/

File: docs/API.md [hunk 1]
```diff
@@ -2417,12 +2417,17 @@ time.ParseDuration function (e.g. "8h", "5m3s").
...
+type UploadInfoResponse struct {
 	// UploadId holds the id of the upload.
 	UploadId string
 
-	// Expires holds when the upload id expires (encoded
-	// in RFC3339 format).
+	// Parts holds all the known parts of the upload.
+	// Parts that haven't been uploaded yet will have nil
+	// elements.
+	Parts Parts
```
Comment (rogpeppe) [inline docs/API.md:2427]: We should probably include the Parts definition in this document too.

File: docs/API.md [hunk 2]
```diff
@@ -2501,13 +2507,28 @@ The response holds a JSON object containing information about the upload.
 
 ```go
 type UploadInfoResponse struct {
```
Comment (rogpeppe) [inline docs/API.md:2525]: Perhaps just refer to the POST /upload endpoint rather than duplicating the type definition?

File: docs/API.md [hunk 3]
```diff
@@ -2417,12 +2417,17 @@ time.ParseDuration function (e.g. "8h", "5m3s").
...
-	// Expires holds when the upload id expires (encoded
-	// in RFC3339 format).
+	// Parts holds all the known parts of the upload.
+	// Parts that haven't been uploaded yet will have nil
+	// elements.
+	Parts Parts
+
+    // Expires holds when the upload id expires (encoded
+    // in RFC3339 format).
 	Expires time.Time
```
Comment (rogpeppe) [inline docs/API.md:2431]: Please add the new offset parameter to the PUT /upload/uploadid section.

**Review by rogpeppe — APPROVED** [review]
LGTM modulo a few minor suggestions below, thanks!

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/charmstore/2094/

**Comment by rogpeppe** [comment]
This was done in PR #792.

**Pull request closed without merging**